parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -103,12 +103,21 @@ write_files:
     {{GetVariableProperty "cloudInitData" "reconcilePrivateHostsService"}}
 {{- end}}
 
+ {{- if IsKrustlet}}
+- path: /etc/systemd/system/krustlet.service
+  permissions: "0644"
+  encoding: gzip
+  owner: root
+  content: !!binary |
+    {{GetVariableProperty "cloudInitData" "krustletSystemdService"}}
```

Comment:
krustlet-fix-ca.sh needs to be written to disk as part of cloud init here, you can use 0544 for the permissions (owner read + execute, others read)